### PR TITLE
Stop and restart color streaming when changing the VideoMode

### PR DESCRIPTION
**Pull request by smits**
I noticed that if the camera is started in 320x240@30Hz it would fail to go to 640x480@30, this PR is fixing that.

### CODE REVIEW

File: openni2_camera/src/openni2_driver.cpp [hunk 1]
```diff
@@ -245,7 +245,15 @@ void OpenNI2Driver::setColorVideoMode(const OpenNI2VideoMode& color_video_mode)
   {
     if (color_video_mode != device_->getColorVideoMode())
     {
+      if (device_->isColorStreamStarted())
+      {
+        device_->stopColorStream();
+      }
       device_->setColorVideoMode(color_video_mode);
+      if (pub_color_.getNumSubscribers() > 0)
```
Comment (adolfo-rt) [inline openni2_camera/src/openni2_driver.cpp:253]: For my understanding, this check on the number of subscribers has the purpose of:
- A small optimization.
- Removes warning messages.
- Prevents an actual error.

**Review by adolfo-rt — APPROVED** [review]
Code LGTM. Untested.

**Review by sdeleersnijder — APPROVED** [review]
LGTM if it fixes the issue

**Review by BrunoChampagne — APPROVED** [review]
I will test it after merge.

File: openni2_camera/src/openni2_driver.cpp (hunk 1 above)
Comment (smits) [reply openni2_camera/src/openni2_driver.cpp:253]: The original code uses this check to start the colorStream, as an optimisation. See https://github.com/Intermodalics/openni2_camera/blob/ros1/openni2_camera/src/openni2_driver.cpp#L386

**Pull request merged**